apps/federatedfilesharing/src/external.js

```diff
@@ -35,7 +35,7 @@ window.OCA.Sharing.showAddExternalDialog = function(share, passwordProtected, ca
 		.replace(/\/$/, '') // remove trailing slash
 
 	showRemoteShareDialog(name, owner, remote, passwordProtected)
-		.then((result, password) => callback(result, { ...share, password }))
+		.then(({ accepted, password }) => callback(accepted, { ...share, password }))
```

Comment:
Why not just

```suggestion
		.then((password) => callback(true, { ...share, password }))
```

so no further changes are needed?

Reply:
It was my initial thought, and then I felt like you had a reason to include the status, and it should be the other way around with the type safety included